common/lib/dependabot/config.rb

```diff
@@ -54,10 +60,22 @@ def initialize(config)
       def ignored_versions_for(dep)
         return [] unless @config[:ignore]
 
-        @config[:ignore].
-          select { |ic| ic[:"dependency-name"] == dep.name }. # FIXME: wildcard support
-          map { |ic| ic[:versions] }.
-          flatten
+        ignore_configs = @config[:ignore].select { |ic| ic[:"dependency-name"] == dep.name } # FIXME: wildcard support
```

Comment:
I started adding a new interface to be able to do this into the update checker https://github.com/dependabot/dependabot-core/pull/3521 

Thinking we probably want to transform the attributes to be `snake_case` ruby symbols so we turn `:'dependency-name'` into `:dependency_name`.

Reply:
> Thinking we probably want to transform the attributes to be snake_case ruby symbols so we turn :'dependency-name' into :dependency_name.

We're using [`symbolize_names`](https://github.com/dependabot/dependabot-core/blob/0352d543c2ed09ed24e42a72d6b0477b23a55dcf/common/lib/dependabot/config/file.rb#L46), is there a similar option to snake case the keys? (I could not find it).

Reply:
We chatted about this and landed on https://github.com/dependabot/dependabot-core/pull/3525 ; I'm going to resolve this thread and force-push a branch that re-does this experiment against that model.